core/src/main/scala/org/bitcoins/core/protocol/ln/currency/MilliSatoshis.scala

```diff
@@ -62,27 +63,63 @@ sealed abstract class MilliSatoshis extends NetworkElement {
   }
 
   def ==(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit == ms.toLnCurrencyUnit
+    toBigInt == ms.toBigInt
   }
 
   def !=(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit != ms.toLnCurrencyUnit
+    toBigInt != ms.toBigInt
   }
 
   def >=(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit >= ms.toLnCurrencyUnit
+    toBigInt >= ms.toBigInt
   }
 
   def >(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit > ms.toLnCurrencyUnit
+    toBigInt > ms.toBigInt
   }
 
   def <(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit < ms.toLnCurrencyUnit
+    toBigInt < ms.toBigInt
   }
 
   def <=(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit <= ms.toLnCurrencyUnit
+    toBigInt <= ms.toBigInt
+  }
+
+  def +(ms: MilliSatoshis): MilliSatoshis = {
+    MilliSatoshis(toBigInt + ms.toBigInt)
+  }
+
+  def -(ms: MilliSatoshis): Try[MilliSatoshis] = {
+    minus(ms)
+  }
+
+  def minus(ms: MilliSatoshis): Try[MilliSatoshis] = {
+    Try {
+      minusExc(ms)
+    }
+  }
+
+  def -!(ms: MilliSatoshis): MilliSatoshis = {
+    minusExc(ms)
+  }
+
+  def minusExc(ms: MilliSatoshis): MilliSatoshis = {
+    MilliSatoshis(toBigInt - ms.toBigInt)
+  }
+
+  /**
+    * Acts similarly to how integer division works. I.e.:
+    * {{{
+    *   13 msat / 4 = 3 msat
+    * }}}
+    */
+  def /(divisor: BigInt): MilliSatoshis = {
```

Comment:
I think this is dangerous and we shouldn't support it in the library. It is too easy for library users to screw this up.